filip/semantics/vocabulary/target_statment.py

```diff
@@ -0,0 +1,267 @@
+    target_statements: List['TargetStatement'] = []
+    """The targetstatements that are combined with this targetstatement 
+     (and/or only)"""
+    type: StatementType = StatementType.LEAF
+    """Statement types"""
+
+    def set_target(self, target_iri: str, target_data_value:str = None):
+        """ Set target for this statement and make it a LEAF statement
+
+        Args:
+            target_iri (str): iri of the target (class or datatype iri)
+            target_data_value (str): Value of the targetstatment if no IRI but
+             a fixed default value is given
+
+        Returns:
+            None
+        """
+        self.type = StatementType.LEAF
+        self.target_iri = target_iri
+        self.target_data_value = target_data_value
+
+    def get_all_targets(self) -> List[List[str]]:
+        """Extract possible targets out of statements
+        interpretation: [[a,b], [c]]-> (a and b) or c -> the target needs to
+        have ancestors(or be): (both a anb b) or c
+
+        items inside the inner brackets are connected via and the innerbrackets
+        are all connect over or
+
+        Returns:
+            List[List[str]]
+        """
+        if self.type == StatementType.LEAF:
+            if self.target_data_value is not None:
+                return [[]]
+            else:
+                return [[self.target_iri]]
+        else:
+            collection = []  # form: [ [[]], [[],[]], ..]
+            for statement in self.target_statements:
+                collection.append(statement.get_all_targets())
+
+            if self.type == StatementType.OR:
+                result = []
+                for sublist in collection:  # sublist form: [[],[],...]
+                    result.extend(sublist)
+                return result
+
+            else:  # AND
+                # with and we distribute our lists
+                # example: col= [ [[1],[2]], [[3,4]] ] => [[1,3,4], [2,3,4]]
+                # we build the results in an all in one way, we compute the
+                # number of entries in the final solution
+                # for each _list li we fill the i-th position of all results
+
+                # example: col= [ [[a,b]], [[c],[d]] , [[e],[f]] ] =>
+                # statement: (a and b) and (c or d) and (e or f)
+                # 0 : res = [[], [], [], []]
+                # 1 : res = [[a,b], [a,b], [a,b], [a,b]]
+                # 2 : res = [[a,b,c], [a,b,c], [a,b,d], [a,b,d]]
+                # 3 : res = [[a,b,c,e], [a,b,c,f], [a,b,d,e], [a,b,d,f]]
+
+                result = []  # result form: [[],[],...]
+                lengths = []
+                number_of_entries = 1
+                for sublist in collection:  # sublist form: [[],[],...]
+                    number_of_entries = number_of_entries * len(sublist)
+                    lengths.append(len(sublist))
+
+                # init with empty lists
+                for c in range(number_of_entries):
```

Comment:
be more pythonic. a variable should have at least 3 characters

Reply:
c was used here as name, as the variable is never used in the for loop. And to my knowledge python does not offer a cleaner way to loop a range without variable. But I changed it now to empty